alacritty/src/event.rs

```diff
@@ -1351,6 +1141,149 @@ impl<N: Notify + OnResize> Processor<N> {
             | GlutinEvent::LoopDestroyed => (),
         }
     }
+}
+
+/// The event processor.
+///
+/// Stores some state from received events and dispatches actions when they are
+/// triggered.
+pub struct Processor {
+    cli_options: CliOptions,
+    config: Config,
+}
+
+impl Processor {
+    /// Create a new event processor.
+    ///
+    /// Takes a writer which is expected to be hooked up to the write end of a PTY.
+    pub fn new(config: Config, cli_options: CliOptions) -> Processor {
+        Processor { cli_options, config }
+    }
+
+    /// Run the event loop.
+    pub fn run(
+        &mut self,
+        mut event_loop: EventLoop<Event>,
+        windows: &mut HashMap<WindowId, WindowContext>,
+    ) {
+        let proxy = event_loop.create_proxy();
+        let mut scheduler = Scheduler::new(proxy.clone());
+
+        // NOTE: Since this takes a pointer to the winit event loop, it MUST be dropped first.
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        let mut clipboard = unsafe { Clipboard::new(event_loop.wayland_display()) };
+        #[cfg(any(not(feature = "wayland"), target_os = "macos", windows))]
+        let mut clipboard = Clipboard::new();
+
+        event_loop.run_return(|event, event_loop, control_flow| {
+            if self.config.ui_config.debug.print_events {
+                info!("glutin event: {:?}", event);
+            }
+
+            // Ignore all events we do not care about.
+            if Self::skip_event(&event) {
+                return;
+            }
+
+            match event {
+                // Check for shutdown.
+                GlutinEvent::UserEvent(Event {
+                    window_id: Some(window_id),
+                    event: EventType::Terminal(TerminalEvent::Exit),
+                }) => {
+                    // Remove the closed terminal.
+                    let window_context = match windows.remove(&window_id) {
+                        Some(window_context) => window_context,
+                        None => return,
+                    };
+
+                    // Shutdown if no more terminals are open.
+                    if windows.is_empty() {
+                        // Write ref tests of last window to disk.
+                        if self.config.ui_config.debug.ref_test {
+                            self.write_ref_test_results(
+                                &window_context.terminal.lock(),
+                                &window_context.display,
+                            );
```

Comment:
Reftests are written for the last window closed. I don't think there's any multi-window aware solution that actually makes sense when it comes to writing reftests, so this should work just fine with just a single window active.